src/actions/build/mod.rs

```diff
@@ -0,0 +1,140 @@
+mod compiler_message_parser;
+
+use std::collections::HashMap;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+
+use analysis::AnalysisHost;
+use hyper::Url;
+use ls_types::Diagnostic;
+use serde_json;
+use span::compiler::DiagnosticSpan;
+
+use build::{BuildPriority, BuildQueue, BuildResult};
+use lsp_data::{NotificationMessage, PublishDiagnosticsParams};
+use server::Output;
+
+use self::compiler_message_parser::{CompilerMessageParser, FileDiagnostic, ParseError};
+
+type Results = HashMap<PathBuf, Vec<Diagnostic>>;
+
+pub struct Build {

```

Comment:
I'm not sure if this is just a naming thing or whether this abstraction could be improved, but it seems kind of wrong at the moment. A `Build` is not a single build, it seems to be more like a `BuildManager` or `BuildHandler`, but that has some overlap with the `BuildQueue`. I think I'm not convinced that there is something worth abstracting out of `ActionHandler` here.